### DISCUSSION TITLE
Impact of .net 6 lack of wcf on CSLA

### DISCUSSION THREAD

**Discussion by kcabral817** [MarimerLLC/csla#2364] [Questions]
Folks,

I've been reading a lot about how WCF is no longer supported in the up coming .Net 6 release and was wondering if that will impact those of us who are using CSLA as a windows service?  Is there something that we should know/do?  My application uses a windows service to communication between the business layer on a server and WPF and don't want to get caught flat-footed.

**Comment by rockfordlhotka** [comment]
It probably depends on which WCF services you are using. If you are just using it for hosting, there are alternatives. If you are using advanced parts of the WS-* spec (transactions, routing, etc.) then I don't have a great answer.

There is an [open source WCF project](https://github.com/CoreWCF/CoreWCF) under way, though I don't know how rapidly it is progressing, as I haven't followed it closely.

Modern .NET does make it very easy to host an HTTP endpoint though, and in fact aspnetcore includes the Kestrel server so web sites can "self host". There are security considerations in how it is configured and used, but if all you are looking for is hosting an endpoint, that might be your answer.

That allows you to use the HttpProxy data portal channel, which (honestly) has seen a lot of love and improvement over the past several years and is almost certainly superior to WcfProxy in most regards at this point.

**Reply by rockfordlhotka** [reply]
There are, btw, two issues you should be aware of.

* #1183 - implement a WCF data portal channel using WCF Core (the open source version) when it is ready
* #2361 - remove the WCF data portal channel from #cslanet in CSLA 6
